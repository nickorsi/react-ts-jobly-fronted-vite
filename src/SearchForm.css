.search-bar-form {
  display: flex;
  height: 3rem;
  margin: 15px;
}

.search-bar-form input {
  border-radius: 0.375rem;
  flex-grow: 1;
  padding-left: 15px;
  margin-right: 15px;
}