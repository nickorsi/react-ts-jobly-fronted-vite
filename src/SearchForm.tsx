import React from "react";

function SearchForm() {
    return (
        <div>
            SearchForm Componenet
        </div>
    )
}

export default SearchForm;