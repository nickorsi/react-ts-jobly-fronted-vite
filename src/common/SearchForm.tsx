import React, { useState } from "react";
import './SearchForm.css';


type SearchFormProps = {
    onSearch: ((param:string) => void),
}
/**
 * SearchForm component used to hold a simple text input and button.
 *
 * Takes a callback function onSearch that runs when the user searches. Note the
 * actually search work does not happen in this component, it just retains the
 * state for the search term and passes it upstream.
 *
 * {JobList, CompanyList} -> SearchForm
 */
function SearchForm({onSearch}: SearchFormProps) {
    const [searchTerm, setSearchTerm] = useState("");

    function handleSearch(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
        e.preventDefault();
        setSearchTerm(searchTerm.trim());
        onSearch(searchTerm.trim());
    }
    return (
        <div className="search-bar">
            <form action="GET" className="search-bar-form">
                <input
                    data-testid="search-bar-input"
                    type="text"
                    placeholder="Enter a search term..."
                    value={searchTerm}
                    onChange={(e) => setSearchTerm(e.target.value)}
                />
                <button
                    data-testid="search-bar-submit"
                    className="btn btn-primary"
                    type="submit"
                    onClick={handleSearch}
                >Search</button>
            </form>
        </div>
    )
}

export default SearchForm;