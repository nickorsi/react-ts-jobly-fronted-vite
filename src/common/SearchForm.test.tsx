import {describe, expect, it, vi} from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import SearchForm from "./SearchForm.tsx";

const fn = vi.fn();
const user = userEvent.setup();

describe("Simple Render Tests", () => {
    it("renders without crashing", function () {
        render(<SearchForm onSearch={fn} />);
    });
    it("matches the snapshot", function () {
        const { container } = render(
            <SearchForm onSearch={fn} />
        );

        expect(container).toMatchSnapshot();
    });
});

describe("SearchForm Interaction Tests", () => {
    it("runs the propfunction when the user searches", async function () {
        render(
            <SearchForm onSearch={fn} />
        );

        const searchInput = screen.getByTestId('search-bar-input') as HTMLInputElement;
        const searchSubmit = screen.getByTestId('search-bar-submit');

        await user.type(searchInput, "Test");
        await user.click(searchSubmit);

        expect(fn).toHaveBeenCalledOnce();
        expect(fn.mock.calls).toEqual([['Test']]);
        // Search bar value should be 'Test' and shouldn't be blank
        expect(searchInput.value).toEqual('Test');
    });
});