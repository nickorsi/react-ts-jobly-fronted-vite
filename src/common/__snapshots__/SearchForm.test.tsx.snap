// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple Render Tests > matches the snapshot 1`] = `
<div>
  <div
    class="search-bar"
  >
    <form
      action="GET"
      class="search-bar-form"
    >
      <input
        data-testid="search-bar-input"
        placeholder="Enter a search term..."
        type="text"
        value=""
      />
      <button
        class="btn btn-primary"
        data-testid="search-bar-submit"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
